### PR TITLE
Feature/74 changes

### PR DESCRIPTION
**Pull request by cookersjs**
A couple of tweaks for orthoinference for v74

- Fix to automated skiplist implementation that caused OrphanedEvents en masse due to Reactions not being connected to their parent Pathways. A full orthoinference run has been completed with these changes and I have confirmed that the OrphanEvents count has gone from ~65k Reactions to 0.

- Automatically attach the previous GraphQA report, now that the newer S3 filesystem has been in place for multiple releases.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -71,15 +53,12 @@ pipeline{
...
-								script{
-									withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]){
-										sh "java -Xmx${env.JAVA_MEM_MAX}m -jar target/orthoinference-*-jar-with-dependencies.jar $ConfigFile ${species}"
-									}
+					speciesList = ['mmus', 'rnor', 'cfam', 'btau', 'sscr', 'drer', 'xtro', 'ggal', 'dmel', 'cele', 'ddis', 'spom', 'scer', 'pfal']
+					for (species in speciesList) {
+						stage("Main: Infer ${species}"){
+							script{
+								withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]){
+									sh "java -Xmx${env.JAVA_MEM_MAX}m -jar target/orthoinference-*-jar-with-dependencies.jar $ConfigFile ${species}"
```
Comment (jweiser) [inline Jenkinsfile:62]: Is the change here the removal of the skip list?

File: Jenkinsfile [hunk 2]
```diff
@@ -147,18 +126,22 @@ pipeline{
...
+					sh "gunzip ${prevGraphQAFileName}.gz"
 					emailext (
-						body: "Hello,\n\nThis is an automated message from Jenkins regarding an update for v${currentRelease}. The Orthoinference step has finished running. Attached to this email should be the summary report output by graph-qa. Please compare this with the graph-qa output from the previous release and confirm if they look appropriate with the developer running Release. \n\nThanks!",
+						body: "Hello,\n\nThis is an automated message from Jenkins regarding an update for v${currentRelease}. The Orthoinference step has finished running. Attached to this email should be the summary report output by graph-qa for both v${currentRelease} and v${previousRelease}. Please compare these and confirm if they look appropriate with the developer running Release. \n\nThanks!",
 						to: '$DEFAULT_RECIPIENTS',
 						from: "${env.JENKINS_RELEASE_EMAIL}",
 						subject: "Orthoinference graph-qa for v${currentRelease}",
-						attachmentsPattern: "**/graph-qa/reports/GraphQA_Summary_v${currentRelease}.csv"
-						
+						attachmentsPattern: "**/graph-qa/reports/GraphQA_Summary_v${currentRelease}.csv, **/${prevGraphQAFileName}"
```
Comment (jweiser) [inline Jenkinsfile]: Could the path for the currentGraphQAFileName also get a variable?  I think the symmetry of both having a variable name would help the readability.

File: Jenkinsfile [hunk 3]
```diff
@@ -147,18 +126,22 @@ pipeline{
...
 			}
 		}
-		// This stage emails the contents of the GraphQA_Summary_vXX.csv that is generated by graph-qa for review. 
+		// This stage emails the contents of the GraphQA_Summary_vXX.csv that is generated by graph-qa for review, as well as the GraphQA_Summary_vXX-1 (previous release). 
 		stage('Post: Email graph-qa output'){
 			steps{
 				script{
+					def prevGraphQAFileName = "GraphQA_Summary_v${previousRelease}.csv"
+					def s3PathPrevGraphQA = "${env.S3_RELEASE_DIRECTORY_URL}/${previousRelease}/orthoinference/reports/${prevGraphQAFileName}.gz"
+					sh "aws s3 cp ${s3PathPrevGraphQA} ."
```
Comment (jweiser) [inline Jenkinsfile]: A line of whitespace between the variable definitions and the commands might be good (not mandatory - just a thought).

File: src/main/java/org/reactome/orthoinference/PathwaysInferrer.java [hunk 4]
```diff
@@ -78,17 +78,10 @@ private static void createInferredPathwayHierarchy(GKInstance sourceEventInst) t
...
-				logger.info("Pathway referral is " + diseasePathwayInst + " -- skipping Pathway inference");
-				return;
-			}
 
 			logger.info("Generating inferred Pathway: " + sourcePathwayReferralInst);
 			// Pathways that have been inferred already are skipped, as are Pathways that are only children of the Disease TopLevelPathway.
-			if (alreadyInferred(sourcePathwayReferralInst) || InstanceUtilities.onlyInDiseasePathway(sourcePathwayReferralInst)) {
-				logger.info("Inferred pathway instance {} already exists or only has Disease for a top level pathway", sourcePathwayReferralInst.getExtendedDisplayName());
-			} else {
+			if (sourceInstanceToInferredInstance.get(sourcePathwayReferralInst) == null && !InstanceUtilities.onlyInDiseasePathway(sourcePathwayReferralInst)) {
```
Comment (jweiser) [inline src/main/java/org/reactome/orthoinference/PathwaysInferrer.java]: Maybe the first half of the condition can be wrapped in a descriptive method name like `noInferenceFor(sourcePathwayReferralInst)`

**Review by jweiser — DISMISSED** [review]
Looks good - left some suggestions but no blockers.

**Review by SolomonShorser-OICR — DISMISSED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (cookersjs) [reply Jenkinsfile:62]: Yes, the skiplist was saved as a 'secret' file that would be injected into the jenkins process -- this forced skiplists to be updated by the user, back when we depended on an externally generated skiplist.

File: src/main/java/org/reactome/orthoinference/PathwaysInferrer.java (hunk 4 above)
Comment (cookersjs) [reply src/main/java/org/reactome/orthoinference/PathwaysInferrer.java]: I used `hasNotBeenInferred(sourcePathwayReferralInst)`

**Review by jweiser — APPROVED** [review]
Looks good!

**Pull request merged**